add ability to find notebooks recursively

hamelsmu opened the pull request:
Closes #358 

Note: I made sure that `.ipynb_checkpoints` is ignored by ignoring all paths that contain the substring `/.`, which ignores all hidden directories.

Ready for review: @jph00 

This also partially related to https://github.com/fastai/nbdev/issues/357.

jph00 commented:
This should be optional and default to false, since otherwise it's a breaking change

hamelsmu commented:
ok @jph00 ready for review.  I made the following changes

- made recursive search optional
- updated the nbdev_build_lib CLI interface (I forgot to do that previously for some reason)
- moved the notebook glob search to its own function 
- Updated tests in `showdoc` that incidentally use this function to test something else about showing the signature

cc: @muellerzr

hamelsmu commented:
@jph00 made changes you requested
